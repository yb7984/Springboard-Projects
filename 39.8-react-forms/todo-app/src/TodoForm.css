.TodoForm label{
    color: blue;
}
.TodoForm input{
    width: 60%;
}
.TodoForm button{
    background-color: blue;
    color: white;
}