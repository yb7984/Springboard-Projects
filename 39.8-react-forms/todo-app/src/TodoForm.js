import { useState } from 'react';
import './TodoForm.css';

const TodoForm = ({ updateTodo, id = "", todo = "" }) => {
    const INITIAL_STATE = {
        id: id,
        todo: todo
    };
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handSumit = (e) => {
        e.preventDefault();
        updateTodo(formData);

        setFormData(INITIAL_STATE);
    };
    const handleOnChange = (e) => {
        const { name, value } = e.target;

        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    return (<div className="TodoForm">
        <form onSubmit={handSumit}>
            <label htmlFor={`todo-${id}`} >{id === "" ? "New" : "Edit"} Todo:</label>
            <input type="text" id={`todo-${id}`} name="todo"
                placeholder="Things to do"
                value={formData.todo}
                onChange={(e) => { handleOnChange(e) }}
                required />

            <button>{id === "" ? "Add" : "Update"} Todo</button>
        </form>
    </div>);
}

export default TodoForm;