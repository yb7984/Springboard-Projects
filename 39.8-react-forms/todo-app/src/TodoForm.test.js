import { render, screen } from '@testing-library/react';
import TodoForm from './TodoForm';

it("renders without crashing", function () {
    render(<TodoForm />);
});

it("matches snapshot, show empty input", function () {
    const { asFragment, queryByLabelText } = render(<TodoForm />);
    expect(asFragment()).toMatchSnapshot();

    const inputTodo = queryByLabelText("New Todo:");

    expect(inputTodo).toBeInTheDocument();
    expect(inputTodo.value).toBe("");
});


it("matches snapshot, show empty input when for edit", function () {
    const { asFragment, queryByLabelText } = render(<TodoForm id="1" todo="test todo" />);
    expect(asFragment()).toMatchSnapshot();

    const inputTodo = queryByLabelText("Edit Todo:");

    expect(inputTodo).toBeInTheDocument();
    expect(inputTodo.value).toBe("test todo");
});