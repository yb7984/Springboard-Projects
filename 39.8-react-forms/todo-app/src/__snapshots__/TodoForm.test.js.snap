// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, show empty input 1`] = `
<DocumentFragment>
  <div
    class="TodoForm"
  >
    <form>
      <label
        for="todo-"
      >
        New Todo:
      </label>
      <input
        id="todo-"
        name="todo"
        placeholder="Things to do"
        required=""
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, show empty input when for edit 1`] = `
<DocumentFragment>
  <div
    class="TodoForm"
  >
    <form>
      <label
        for="todo-1"
      >
        Edit Todo:
      </label>
      <input
        id="todo-1"
        name="todo"
        placeholder="Things to do"
        required=""
        type="text"
        value="test todo"
      />
      <button>
        Update Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
